src/scadapi.cc

```diff
@@ -19,6 +19,26 @@ ScadApi::~ScadApi()
 
 void ScadApi::updateAutoCompletionList(const QStringList &context, QStringList &list)
 {
+    //first see if we are in a string literal. if so, don't allow auto comlete
+    bool lastWasEscape=false;
+    bool inString=false;
+    int line, col;
+    qsci->getCursorPosition(&line, &col);
+    auto sLine = qsci->text(line);  //get the current line of text
+    QByteArray ba = sLine.toLocal8Bit();
```

Comment:
Using local 8-bit data breaks the column calculation if the string contains multi-byte UTF-8 characters, e.g. typing:
```
echo("😀😀😀"); c.c.c.
```
will popup the autocomplete only at the 3rd `c.` as each of the [smiley emoji](https://emojipedia.org/grinning-face/) is encoded in 4 bytes (U+1F600 = f0 9f 98 80).

I think for the current logic of iterating though the beginning of the line, converting via `sLine.toStdU32String();` and using `char32_t` should work as 32-bit per character are wide enough to always map 1:1.

Reply:
ok, I thought it would just see the '"' as an 8 bit.

I will make the change.

Reply:
Yes, the quote is handled just fine and there's no possible collision with the utf-8 encoding as multi-byte cases will have the top-most bit set. But the line length is 20 (including the internal newline) but with `toLocal8Bit()` the `QByteArray is actually 29 long. So the closing quote is only seen at position 19.

Reply:
ok, that makes sense. I didn't realize that the local 8 bit routine would return multiple 8 bit characters for multi-byte. 

First time using Qt. Live and learn, die and forget it all. ;-)

Reply:
Yeah, no worries, it's just a thing I'm trying to watch out for as we had a couple of those issues in the past like this dance from #3071 for the search highlight:
https://github.com/openscad/openscad/blob/8d63488af35a46011899ee42a4ee0bb7de7f291c/src/scintillaeditor.cpp#L662-L665

There's still a couple of places where those multi-bytes encodings cause trouble, like the error marker is also off a bit in some cases.

Reply:
It should and it does output smileys on my Linux machine. I just did a fix on the error output so if it still does not work for you with the latest master, that's a bug. I'm not sure, but I think there's a bug report for Windows having issues with UTF-8 in the console.

Reply:
I have a master from a couple of days ago. I can try on the latest later.